exporter/api.py

```diff
@@ -22,28 +23,78 @@ def export_resources_to_directory(learning_resources):
         learning_resources (list of learningresources.models.LearningResource):
             LearningResources to export in tarball
     Returns:
-        unicode: Newly created temp directory with files inside of it.
+        (unicode, bool):
+            First item is newly created temp directory with files inside of it.
+            Second item is True if a static asset collision was detected.
     """
-    def sanitize(title):
+    def sanitize(url_name):
         """Sanitize title for use in filename."""
         # Limit filename to 200 characters since limit is 256
         # and we have a number and extension too.
-        return slugify(title)[:200]
+        return slugify(url_name)[:200]
 
     tempdir = mkdtemp()
+    collision = False
     try:
         for learning_resource in learning_resources:
-            filename = "{id}_{title}.xml".format(
-                id=learning_resource.id,
-                title=sanitize(learning_resource.title),
-            )
-            with open(os.path.join(tempdir, filename), 'w') as f:
+            if learning_resource.url_name is not None:
+                filename = "{id}_{url_name}.xml".format(
+                    id=learning_resource.id,
+                    url_name=sanitize(learning_resource.url_name),
+                )
+            else:
+                filename = "{id}.xml".format(id=learning_resource.id)
+
+            type_name = learning_resource.learning_resource_type.name
+            try:
+                os.mkdir(os.path.join(tempdir, type_name))
+            except OSError as ex:
+                # Directory already exists
+                if "File exists" not in ex.args:
+                    raise
+            with open(os.path.join(tempdir, type_name, filename), 'w') as f:
                 f.write(learning_resource.content_xml)
+
+            for static_asset in learning_resource.static_assets.all():
+                try:
+                    os.mkdir(os.path.join(tempdir, "static"))
+                except OSError as ex:
+                    # Directory already exists
+                    if "File exists" not in ex.args:
+                        raise
+
+                # Output to static directory.
+                static_filename = os.path.basename(static_asset.asset.name)
+
+                def static_file_path(name):
+                    """Convenience function for making a static path."""
+                    return os.path.join(tempdir, "static", name)
+
+                if os.path.exists(static_file_path(static_filename)):
+                    collision = True
+
+                    base, ext = os.path.splitext(static_filename)
+
+                    i = 1
+                    while True:
```

Comment:
Minor style difference; I might make the loop `while os.path.exists(static_file_path(static_filename)):` instead. It's a little simpler, and `while True` makes me nervous.